### PR TITLE
Refactor discretization.

### LINKED ISSUE #1408: Move `cv_policy` out of `cable_cell_parameter_set`
The `cv_policy` of a `cable_cell` is currently exposed through `cable_cell_parameter_set`, which is a member of the `decor` and `cable_cell_global_properties`. This allows users to set the `cv_policy` at the cell and model levels respectively. 
The issue is that the `cv_policy` is an implementation detail and doesn't really belong in the `decor` of a cell. Now that we're working on a cable-cell format (#1336), it would be a good time to refactor `cable_cell_parameter_set` to not include the `cv_policy` and have the discretization info relayed to the `cable_cell` in another way.

### PR DESCRIPTION
**Pull request by thorstenhater**
This PR rectifies a long-standing issue since the introduction of `decor`: 
CV discretization was treated as a defaultable value instead of a cell property, along with bio-physical properties.

This lead to some awkward facts:
- Intermingling of bio-physics and numerical settings
- Discretization is treated like a default without override.
- Discretization cannot be changed once set, e.g. if read from `.acc`, requiring the copy+update of a decor instead

Thus:
- Discretization can now be set during `cable_cell` construction
- It can be read and updated later
- Refactor `cv_policy` to use type erasure like other, similar constructions, e.g. schedules.
- Hide internal `cv_policy` constructors.

⚠️ Breaks `.acc` by upgrading from 0.9 to 0.10 ⚠️ 

Closes #1408

### CODE REVIEW

File: python/strprintf.hpp [hunk 1]
```diff
@@ -168,7 +168,7 @@ namespace impl {
             for (auto& x: s.seq_) {
                 if (!first) o << s.sep_;
                 first = false;
-                o << s.f(x);
+                o << s.f_(x);
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply python/strprintf.hpp:171]: Yes, caught by new clang, has been in there forever...

File: python/example/single_cell_bluepyopt_l5pc.py [hunk 2]
```diff
@@ -21,58 +20,54 @@
...
-# (6) Create a class that inherits from arbor.recipe
-class single_recipe(arbor.recipe):
+# (6) Create a class that inherits from A.recipe
+class single_recipe(A.recipe):
     # (6.1) Define the class constructor
     def __init__(self, cell, probes):
         # The base C++ class constructor must be called first, to ensure that
         # all memory in the C++ class is initialized correctly.
-        arbor.recipe.__init__(self)
+        A.recipe.__init__(self)
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply python/example/single_cell_bluepyopt_l5pc.py:53]: I had a vague memory that this is frowned upon, but cannot back it up anymore. 
Your suggestion is certainly more in line w/ OOP principles.
Given that, we should make a note to change it everywhere.

File: arbor/cable_cell.cpp [hunk 3]
```diff
@@ -232,8 +240,8 @@ void cable_cell_impl::init() {
     }
 }
 
-cable_cell::cable_cell(const arb::morphology& m, const decor& decorations, const label_dict& dictionary):
-    impl_(make_impl(new cable_cell_impl(m, dictionary, decorations)))
+cable_cell::cable_cell(const arb::morphology& m, const decor& decorations, const label_dict& dictionary, const std::optional<cv_policy>& cvp):
+    impl_(make_impl(new cable_cell_impl(m, dictionary, decorations, cvp)))
```
[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/cable_cell.cpp:244]: Yes, I wondered about that while passing over it. I'll change it.

[the remark this answers is not in the document]
Comment (thorstenhater) [reply arbor/cable_cell.cpp:244]: No ;) Problem here is that `_impl` has an incomplete type, so we cannot use `default_delete` here.
Clang is helpful enough to point that out:
```
/opt/homebrew/Cellar/llvm/19.1.1/bin/../include/c++/v1/__memory/unique_ptr.h:78:19: error: invalid application of 'sizeof' to an incomplete type 'arb::cable_cell_impl'
   78 |     static_assert(sizeof(_Tp) >= 0, "cannot delete an incomplete type");
```

**Review by boeschf — APPROVED** [review]

File: arbor/cable_cell.cpp (hunk 3 above)
[the remark this answers is not in the document]
Comment (boeschf) [reply arbor/cable_cell.cpp:244]: ok :)

**Pull request merged**